diff --git a/arch/powerpc/Makefile b/arch/powerpc/Makefile
index 11a1acba164a..aca24497acb1 100644
--- a/arch/powerpc/Makefile
+++ b/arch/powerpc/Makefile
@@ -24,8 +24,8 @@ ifdef CONFIG_PPC32
 # or platform code sets later on the command line, but they are needed
 # to set a sane 32-bit cpu target for the 64-bit cross compiler which
 # may default to the wrong ISA.
-KBUILD_CFLAGS		+= -mcpu=powerpc
-KBUILD_AFLAGS		+= -mcpu=powerpc
+KBUILD_CFLAGS		+= -mcpu=8548
+KBUILD_AFLAGS		+= -mcpu=8548
 endif
 endif
 endif
@@ -85,7 +85,7 @@ else
 KBUILD_CFLAGS += $(call cc-option,-mbig-endian)
 KBUILD_LDFLAGS	+= -EB
 LDEMULATION	:= ppc
-GNUTARGET	:= powerpc
+GNUTARGET	:= powerpc3
 MULTIPLEWORD	:= -mmultiple
 endif
 
@@ -157,7 +157,7 @@ CFLAGS-$(CONFIG_GENERIC_CPU) += $(call cc-option,-mtune=power7,$(call cc-option,
 CFLAGS-$(CONFIG_GENERIC_CPU) += $(call cc-option,-mcpu=power5,-mcpu=power4)
 endif
 else
-CFLAGS-$(CONFIG_GENERIC_CPU) += -mcpu=powerpc64
+CFLAGS-$(CONFIG_GENERIC_CPU) += -mcpu=8548
 endif
 
 ifdef CONFIG_MPROFILE_KERNEL
@@ -168,18 +168,18 @@ CFLAGS-$(CONFIG_TARGET_CPU_BOOL) += $(call cc-option,-mcpu=$(CONFIG_TARGET_CPU))
 
 # Altivec option not allowed with e500mc64 in GCC.
 ifdef CONFIG_ALTIVEC
-E5500_CPU := -mcpu=powerpc64
+E5500_CPU := -mcpu=powerpc642
 else
-E5500_CPU := $(call cc-option,-mcpu=e500mc64,-mcpu=powerpc64)
+E5500_CPU := $(call cc-option,-mcpu=e500mc64,-mcpu=powerpc643)
 endif
 CFLAGS-$(CONFIG_E5500_CPU) += $(E5500_CPU)
 CFLAGS-$(CONFIG_E6500_CPU) += $(call cc-option,-mcpu=e6500,$(E5500_CPU))
 
 ifdef CONFIG_PPC32
 ifdef CONFIG_PPC_E500MC
-CFLAGS-y += $(call cc-option,-mcpu=e500mc,-mcpu=powerpc)
+CFLAGS-y += $(call cc-option,-mcpu=e500mc,-mcpu=powerpc4)
 else
-CFLAGS-$(CONFIG_E500) += $(call cc-option,-mcpu=8540 -msoft-float,-mcpu=powerpc)
+CFLAGS-$(CONFIG_E500) += $(call cc-option,-mcpu=8540 -msoft-float,-mcpu=8548)
 endif
 endif
 
@@ -226,7 +226,7 @@ KBUILD_CFLAGS	+= $(call cc-option,-fno-dwarf2-cfi-asm)
 KBUILD_CFLAGS		+= $(call cc-option,-mno-string)
 
 ifdef CONFIG_6xx
-KBUILD_CFLAGS		+= -mcpu=powerpc
+KBUILD_CFLAGS		+= -mcpu=powerpc6
 endif
 
 # Work around a gcc code-gen bug with -fno-omit-frame-pointer.
