diff -ruNp Python-3.7.4.orig/configure.ac Python-3.7.4/configure.ac
--- Python-3.7.4.orig/configure.ac	2019-09-04 15:54:17.606209722 +0200
+++ Python-3.7.4/configure.ac	2019-09-04 15:55:23.666291574 +0200
@@ -726,6 +726,13 @@ fi
 
 
 MULTIARCH=$($CC --print-multiarch 2>/dev/null)
+
+# Tweak MULTIARCH
+if test x$MULTIARCH = xpowerpc-linux-gnuspev1
+then
+	MULTIARCH="powerpc-linux-gnuspe"
+fi
+
 AC_SUBST(MULTIARCH)
 
 AC_CANONICAL_TARGET
